docs/vc-spec.md

```diff
@@ -0,0 +1,313 @@
+
+### 1: Load II in a new window
+
+The II window needs to be opened with the subpath `/vc-flow`.
+
+After opening the II window, II will load and notify `window.opener` with the following JSON-RPC notification:
+
+```json
+{
+  "jsonrpc": "2.0",
+  "method": "vc-flow-ready"
+}
+```
+
+### 2: Request a VC
+
+After receiving the notification that II is ready, the relying party can request a VC by sending the following JSON-RPC request:
+ 
+* Method: `request_credential`
+* Params:
+  * `issuer`: An issuer that the relying party trusts. It has the following properties:
+    * `origin`: The origin of the issuer.
+    * `canisterId`: (optional) The canister id of the issuer, if applicable/known.
+  * `credentialSpec`: The spec of the credential that the relying party wants to request from the issuer.
+    * `credentialName`: The name of the requested credential.
+    * `arguments`: (optional) A map with arguments specific to the requested credentials. It maps string keys to values that must be either strings or integers.
+  * `credentialSubject`: The subject of the credential as known to the relying party. Internet Identity will use this principal to ensure that the flow is completed using the matching identity.
+
+#### Examples
+
+```json
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "method": "request_credential",
+  "params": {
+    "issuer": {
+        "origin": "https://employment-info.com",
+        "canisterId": "rwlgt-iiaaa-aaaaa-aaaaa-cai"
+    },
+    "credentialSpec": {
+        "credentialName": "VerifiedEmployee",
+        "arguments": {
+          "employerName": "XYZ Ltd."
+        }
+    },
+    "credentialSubject": "2mdal-aedsb-hlpnv-qu3zl-ae6on-72bt5-fwha5-xzs74-5dkaz-dfywi-aqe"
+  }
+}
+```
+
+```json
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "method": "request_credential",
+  "params": {
+    "issuer": {
+        "origin": "https://kyc-star.com",
+        "canisterId": "rdmx6-jaaaa-aaaaa-aaadq-cai"
+    },
+    "credentialSpec": {
+        "credentialName": "VerifiedAdult",
+        "arguments": {
+            "minAge": 21
+        }
+    },
+    "credentialSubject": "s33qc-ctnp5-ubyz4-kubqo-p2tem-he4ls-6j23j-hwwba-37zbl-t2lv3-pae"
+  }
+}
+```
+
+```json
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "method": "request_credential",
+  "params": {
+    "issuer": {
+        "origin": "https://kyc-resident-info.org"
+    },
+    "credentialSpec": {
+        "credentialName": "VerifiedResident",
+        "arguments": {
+            "countryName": "Panama",
+            "countryAlpha2": "PA"
+        }
+    },
+    "credentialSubject": "cpehq-54hef-odjjt-bockl-3ldtg-jqle4-ysi5r-6bfah-v6lsa-xprdv-pqe"
+  }
+}
+```
+
+### 3: Get a Response
+
+
+#### Receive a Verifiable Presentation
+
+After the user has successfully completed the flow, Internet Identity will respond with the following JSON-RPC response:
```

Comment:
Excerpt of offline discussion:

> Q: does the order of credentials (alias & actual credentials) matter? assuming it doesn’t since the spec doesn’t mention it

> A: The spec doesn't mandate the ordering explicitly, but the bullet points for the returned presentation list the id alias credential first, so let's stick to this ordering (I'll add to the spec an explicit note for this ordering, thanks for pointing this out!)